src/test/testUtil.ts

```diff
@@ -136,21 +137,30 @@ export function installFakeClock(): FakeTimers.InstalledClock {
  * Unlike {@link assertTelemetry}, this function does not do any transformations to
  * handle fields being converted into strings. It will also not return `passive` or `value`.
  */
-export function getMetrics<K extends MetricName>(
-    name: K,
-    ...excludeKeys: string[]
-): readonly Partial<MetricShapes[K]>[] {
-    const query = { metricName: name, excludeKeys: ['awsAccount', 'awsRegion', ...excludeKeys] }
+export function getMetrics<K extends MetricName>(name: K): readonly Partial<MetricShapes[K]>[] {
+    const query = { metricName: name }
 
     return globals.telemetry.logger.query(query) as unknown as Partial<MetricShapes[K]>[]
 }
 
+// Note that Typescript is unable to describe the set of all supertypes of a type.
+// This should be typed something like `asserts actual is * extends T`
+export function partialDeepCompare<T>(actual: unknown, expected: T, message?: string): asserts actual is T {
+    if (typeof actual !== 'object' || !actual || typeof expected !== 'object' || !expected) {
+        assert.deepStrictEqual(actual, expected, message)
+    }
+
+    const partial = selectFrom(actual, ...keys(expected as object))
+    assert.deepStrictEqual(partial, expected, message)
+}
+
 /**
  * Finds the emitted telemetry metrics with the given `name`, then checks if the metadata fields
- * match the expected values, in the (non-continguous) order specified by `expected`.
+ * match the expected values, in the order specified by `expected`. Comparisons are done using
+ * {@link partialDeepCompare}. **Only fields present in {@link expected} will be checked.**
  *
  * @param name Metric name
- * @param expected Metric(s) shape(s) which are compared _in order_ (non-continguous) to metrics matching `name`.
```

Comment:
@justinmk3 The non-contiguous comparison wasn't working correctly for me. Was it originally added to deal with flakiness?

Reply:
it was added so that the caller didn't need to care about the exact sequence of metrics emitted

Reply:
Ah ok, maybe some other change was causing issues. I'll try to add it back because it would be useful for testing sequences of events with different names.